File: fadingWind/fadingWind.cpp
#include <iostream>
#include <algorithm>

using namespace std;

int main() {
  int h,k,v,s;
  cout << "Enter the numbers: ";
  cin >> h >> k >> v >> s;

  int dist = 0;

  while (h > 0) {
    v += s;
    v -= max(1, v / 10);
    if (v >= k) { h++;}
    if (0 < v && v < k) {
      h--;
      if (h == 0) {v = 0;}
    }
    if (v <= 0) {h = 0; v = 0;}
    dist += v;
    if (s > 0) {s--;}
  }

  cout << dist << endl;
  return dist;
}

File: fadingWind/fadingWind.py
def main():
    h,k,v,s = input("Enter 4 integers: ").split()
    h = int(h); k = int(k); v = int(v); s = int(s)
    dist = 0
    while h > 0:
        v += s
        v -= max(1, int(v/10))
        if v >= k:
            h += 1
            if h == 0:
                v = 0
        if v <= 0:
            h = 0; v = 0
        dist += v
        if s > 0:
            s -= 1
    print(dist)

if __name__ == "__main__":
    main()
